src/tools/compiletest/src/runtest.rs

```diff
@@ -2208,7 +2209,17 @@ impl<'test> TestCx<'test> {
 
     fn fatal_proc_rec(&self, err: &str, proc_res: &ProcRes) -> ! {
         self.error(err);
-        proc_res.fatal(None);
+        proc_res.fatal(None, || ());
+    }
+
+    fn fatal_proc_rec_with_ctx(
```

Comment:
Do we even need this as a function, I believe we could just inline this right since it is only used in one place?

Reply:
If this is inline, it's really annoying to work with:

- you have to use `this` instead of `self` to avoid borrowck errors
- all the code is indented like 4 levels